### PR TITLE
feat: added run python

### PR DESCRIPTION
**Pull request by h4ck3rk3y**
## Description:
Added run python so that users can easily run python scripts

## Is this change user facing?
YES

### CODE REVIEW

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/tasks_shared.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+	if waitValueStr, ok := waitValue.(starlark.String); ok {
+		waitTimeout = waitValueStr.GoString()
+	} else if _, ok := waitValue.(starlark.NoneType); ok {
+		waitTimeout = DisableWaitTimeoutDurationStr
+	}
+	return waitTimeout, nil
+}
+
+func createInterpretationResult(resultUuid string, fileArtifactNames []string) *starlarkstruct.Struct {
+	runPythonCodeValue := fmt.Sprintf(magic_string_helper.RuntimeValueReplacementPlaceholderFormat, resultUuid, runResultCodeKey)
```
Comment (Peeeekay) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/tasks_shared.go]: can we rename this to `runCodeValue` and `runOutputValue`

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/tasks_shared.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+		filesArtifactExpansion,
+		0,
+		0,
+		service_config.DefaultPrivateIPAddrPlaceholder,
+		0,
+		0,
+		// TODO: hardcoding subnetwork to default is what we do now but is incorrect as the run_sh might not be able to
+		//  reach some services outside of the default subnetwork. It should be re-worked if users want to use that in
+		//  conjunction with subnetworks
+		service_config.DefaultSubnetwork,
```
Comment (Peeeekay) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/tasks_shared.go:254]: oh good catch.

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_python.go [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+			return nil, startosis_errors.WrapWithInterpretationError(err, "Unable to extract value for '%s' argument", ImageNameArgName)
+		}
+		image = imageStarlark.GoString()
+	} else {
+		image = defaultRunPythonImageName
+	}
+
+	var filesArtifactExpansion *files_artifacts_expansion.FilesArtifactsExpansion
+	if arguments.IsSet(FilesArgName) {
+		filesStarlark, err := builtin_argument.ExtractArgumentValue[*starlark.Dict](arguments, FilesArgName)
```
Comment (Peeeekay) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_python.go:206]: you could potentially refactor this one too; as you did for others.

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_python.go [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+			return nil, stacktrace.Propagate(err, "an error occurred while replacing runtime value in a package passed to run_python")
+		}
+		maybePackagesWithRuntimeValuesReplaced = append(maybePackagesWithRuntimeValuesReplaced, maybePackageWithRuntimeValueReplaced)
+	}
+
+	if len(maybePackagesWithRuntimeValuesReplaced) == 0 {
+		return nil, nil
+	}
+
+	packageInstallationSubCommand := fmt.Sprintf("pip install %v", strings.Join(maybePackagesWithRuntimeValuesReplaced, spaceDelimiter))
```
Comment (Peeeekay) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_python.go]: can we have this as const: `pip install`

**Review by Peeeekay — APPROVED** [review]
looks good; test are passing - left small comments.

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_python.go (hunk 3 above)
Comment (h4ck3rk3y) [reply core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_python.go:206]: they're slightly different as one sets the python file in addition to the defaults while the other doesn't! will leave it as is.

**Pull request merged**